nucypher/crypto/powers.py

```diff
@@ -67,69 +76,159 @@ def consume_power_up(self, power_up):
     """
     Allows for transacting on a Blockchain via web3 backend.
     """
-    not_found_error = NoBlockchainPower
+    not_found_error = NoTransactingPower
+
+    class NoBlockchainConnection(PowerUpError):
+        pass
+
+    class AccountLocked(PowerUpError):
+        pass
 
-    def __init__(self, blockchain: 'Blockchain', account: str, device = None) -> None:
+    class InvalidSigningRequest(PowerUpError):
+        pass
+
+    def __init__(self,
+                 blockchain,
+                 account: str,
+                 password: str = None,
+                 device=NO_STAKING_DEVICE):
         """
-        Instantiates a BlockchainPower for the given account id.
+        Instantiates a TransactingPower for the given checksum_address.
         """
+        if password and (device is not NO_STAKING_DEVICE):
+            raise ValueError(f"Cannot create a {self.__class__.__name__} with both a client and an device signer.")
+
         self.blockchain = blockchain
+
+        if blockchain.is_connected:
+            self.client = blockchain.client
+        else:
+            self.client = NO_BLOCKCHAIN_CONNECTION
+
         self.account = account
         self.device = device
-        self.is_unlocked = False
 
-    def unlock_account(self, password: str):
-        """
-        Unlocks the account for the specified duration. If no duration is
-        provided, it will remain unlocked indefinitely.
-        """
-        self.is_unlocked = self.blockchain.client.unlock_account(self.account, password)
-        if not self.is_unlocked:
-            raise PowerUpError("Failed to unlock account {}".format(self.account))
+        self.__activated = False
+        self.__password = password
+        self.__unlocked = False
+
+    def __del__(self):
+        self.lock_account()
+
+    @property
+    def is_unlocked(self) -> bool:
+        return self.__unlocked
+
+    @property
+    def is_active(self) -> bool:
+        """Returns True if the blockchain currently has this transacting power attached."""
+        return self.blockchain.transacting_power == self
+
+    def activate(self, password: str = None):
+        """Be Consumed"""
+        self.blockchain.connect()
+        self.client = self.blockchain.client       # Connect
+        self.unlock_account(password=password)     # Unlock
+        self.blockchain.transacting_power = self   # Attach
+        self.__password = None                     # Discard
+        self.__activated = True                    # Remember
+
+    def lock_account(self):
+        if self.device is not NO_STAKING_DEVICE:
+            # TODO: Implement TrustedDevice
+            _result = self.device.lock()
+        else:
+            _result = self.client.lock_account(address=self.account)
+        self.__unlocked = False
+
+    def unlock_account(self, password: str = None):
+        if self.device is not NO_STAKING_DEVICE:
+            # TODO: Embed in TrustedDevice
+            ping = 'PING|PONG'
```

Comment:
Is it how Trezor pings, or ping function just repeats any message?

Reply:
Any message will be returned by the trezor here.